data-model/src/main/avro/RawEventLog.avdl

```diff
@@ -0,0 +1,19 @@
+@namespace("org.hypertrace.core.datamodel")
+protocol LogEventRecordsProtocol {
+  import idl "Attributes.avdl";
+
+  record LogEventRecord {
```

Comment:
should we make all of these nullable/defaultable? I feel like this came up but don't remember where it landed. Functionally at least, span and trace id eventually will be, while time stamp and tenant id never will be.

Reply:
Agree for time stamp & tenant id.
I am trying to understand would it make sense to say store a record with null spanId/traceId/attributes?
From our use case I feel spanId/traceId/attributes should be required fields.

Reply:
I think there's two considerations
- Avro compatibility in general: From some PR comments with @avinashkolluru , I was under the impression that we are _always_ defaulting avro records because it makes things easier to work with (and matches how our other IDL, proto, works), and enforcing required schema constraints in code.
- This particular schema: spanId and traceId will always come in for our current use case (embedded logs), but this overall design is also in part accounting for the standalone log case, right? We know that case will have more requirements, but breaking the ones we already know of doesn't seem necessary. Attributes - even for our current use case, having no attributes is valid - a log event need not contain any. We could model that as an empty attributes object, so this case kind of goes back to the first compatibility consideration.

Reply:
For spanId/traceId, I thought the conclusion was to support only those standalone logs which have valid spanId/traceId in it. Anyways, making it optional will keep things flexible. 

Between nullable and default value what you would suggest to use for the optional fields?

Reply:
> I thought the conclusion was to support only those standalone logs which have valid spanId/traceId in it. Anyways, making it optional will keep things flexible.

I definitely may have misunderstood, but I thought that was just the an iterative goal towards full log support.

> Between nullable and default value what you would suggest to use for the optional fields?

Depends on the fields and their meaning. For something like a spanId, there's no meaningful default, so that should be nullable. For timestamp, there's also no meaningful default, but I'm not sure how avro generates that for a primitive - if it doesn't make it harder to use or more susceptible to errors, I'd go null - else a default. For attributes, that's another object underneath there, but if it allows it, defaulting that would make it easy to use such that we don't have two different states to represent empty.

Reply:
> Between nullable and default value what you would suggest to use for the optional fields?

For a string, the default usually is null. But avro again has this weirdness where you can't assign a null value to a string. So you need a union like: `union { null, string } some_field = null;`